### PR TITLE
Tests for Loss functions

### PR DESCRIPTION
**Pull request by charleygros**
This PR aims at introducing tests for our loss functions.

Note: this PR is aligned with #256.

DONE:
- test_multiclassdiceloss
- test_generaliseddiceloss
- Change GDL sign so that it is consistent with the other losses, ie we want to minimize the loss: best=-1, worst=0
- test_tverskyloss
- test_focaltverskyloss
- test_diceloss (cf #230 )
- test_bce

### CODE REVIEW

**Comment by charleygros** [comment]
The beauty of GDL is:

> weighting,  the  contribution  of  each  label  iscorrected by the inverse of its volume, thus reducing the well known correlationbetween region size and Dice score

... --> our implementation is useless!
Indeed, from our implementation perspective we have only one class (one channel) --> only one weight is computed (the one of the SC/tumor/lesion) --> our GDL is basically the DiceLoss.

I have added a parameter (`include_background`) which essentially add a channel to the input tensor reprensenting the background class (ie 1 for background, 0 for foreground). Thus, the weighting between classes is correctly done.

```python
        if self.include_background:
            # init
            input_background = torch.zeros(input.size(), dtype=input.dtype)
            target_background = torch.zeros(target.size(), dtype=target.dtype)
            # fill with opposite
            input_background[input == 0] = 1
            target_background[target == 0] = 1
            # Concat
            input = torch.cat([input, input_background], dim=1)
            target = torch.cat([target, target_background], dim=1)
```

**Comment by charleygros** [comment]
All tests receive tuples organised as follows: input tensor, target tensor, expected_value, loss function.

File: ivadomed/losses.py [hunk 1]
```diff
@@ -112,27 +112,44 @@ class GeneralizedDiceLoss(nn.Module):
...
+            input_background[input.sum(1).expand_as(input_background) == 0] = 1
+            target_background[target.sum(1).expand_as(input_background) == 0] = 1
+            # Concat
+            input = torch.cat([input, input_background], dim=1)
+            target = torch.cat([target, target_background], dim=1)
+
+        # Compute class weights
         target = target.float()
-        target_sum = target.sum(-1)
+        target_sum = target.sum(-1).sum(-1)
```
Comment (andreanne-lemay) [inline ivadomed/losses.py:143]: Here im guessing in 3D we'd want .sum(-1).sum(-1).sum(-1)?

File: testing/test_losses.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+# pytest unit tests for ivadomed.losses
+
+import torch
```
Comment (andreanne-lemay) [inline testing/test_losses.py:5]: Great test :100:
